File: LeetCode/Problems/C++/161. One Edit Distance.cpp
class Solution {
public:
    bool isOneEditDistance(string s, string t) {
        if(!s.size() && !t.size()) return false;
        
        int n_s = s.size();
        int n_t = t.size();
        
        if(n_s > n_t) 
            return isOneEditDistance(t,s);
        
        if(n_t - n_s > 1) return false;
        
        int i = 0, j = 0;
        while(i < n_s){
            if(s[i]!=t[j]){
                if(n_s == n_t) return s.substr(i+1,n_s-i-1) == t.substr(j+1,n_t-j-1);
                else return s.substr(i,n_s-i) == t.substr(j+1,n_t-j);
            }
            else{
                i++;
                j++;
            }
        } 
        return n_s +1 == n_t;
    }
};

File: LeetCode/Explore/Queue & Stack/161. One Edit Distance.py
#Time: O(max(m,n))
#Space: O(1)
class Solution:
    def isOneEditDistance(self, s: str, t: str) -> bool:
        edited = False
        
        ptr_s = 0
        ptr_t = 0
        if s == t or abs(len(s) - len(t)) >=2:
            return False
        
        while ptr_s < len(s) and ptr_t < len(t):
            if s[ptr_s] != t[ptr_t]:
                if not edited:
                    edited = True
                else:
                    return False
                if len(s) > len(t):
                    ptr_t-=1
                if len(s) < len(t):
                    ptr_s-=1
            ptr_s+=1
            ptr_t+=1

        return True